packages/opal-common/opal_common/git/commit_viewer.py

```diff
@@ -210,6 +210,29 @@ def directories(self, filter: Optional[DirectoryFilter] = None) -> Generator[Ver
                 if filter(directory):
                     yield directory
 
+    def get_node(self, path: Path, filterable_gen: Optional[Callable] = None) -> Optional[VersionedNode]:
+        """
+        Returns the node in the given path, None if it doesn't exist
+        """
+        filterable_gen = filterable_gen or self.nodes
+        path = Path(path) # Works even if path is already Path
+        nodes = filterable_gen(lambda n: n.path == path)
```

Comment:
I feel like [`itertools.filterfalse`](https://docs.python.org/3/library/itertools.html#itertools.filterfalse) will help you here -- maybe it can replace this whole method.

Reply:
Not sure how `filterfalse` would help me, but `filter` actually helps.
My code there was unnecessarily complex - I've fixed it and I believe it's much more readable now (with a small price of having a slightly duplicated [line of] code - but I think it's justifiable)